spice_ev/costs.py

```diff
@@ -458,19 +458,32 @@ def calculate_costs(strategy, voltage_level, interval,
             calculate_commodity_costs(price_list_flex_load, power_flex_load_list,
                                       interval, fraction_year)
 
-        # capacity costs for flexible load:
-        power_outside_core_standing_time_flex_list = [
-            p for i, p in enumerate(power_flex_load_list) if p < 0 and not
-            util.dt_within_core_standing_time(timestamps_list[i], core_standing_time_dict)]
+        # DEVIATION COSTS:
+
+        # positive deviation from schedule concerning grid supply (not feed-in):
+        power_grid_supply_schedule_list = [max(v, 0) for v in power_schedule_list]
+        pos_deviation_grid_supply_list = [max(schedule_power - power_grid_supply_list[i], 0)
+                                          for i, schedule_power
+                                          in enumerate(power_grid_supply_schedule_list)]
+
+        # charge for deviation from schedule:
+        schedule_deviation_charge = schedule_charges["deviation_charge"]
+
+        # tollerance for charging deviation from schedule:
+        schedule_deviation_tollerance = schedule_charges["deviation_tollerance"]
+
+        # capacity related costs for deviation from schedule:
+        max_pos_deviation_grid_supply = max(pos_deviation_grid_supply_list)
+        max_grid_supply_schedule = max(power_grid_supply_schedule_list)
+        lower_limit_deviation = max_grid_supply_schedule * schedule_deviation_tollerance
 
-        # charging only within core standing time (cst)
-        if power_outside_core_standing_time_flex_list == []:
-            max_power_grid_supply_outside_cst_flex = 0
+        if max_pos_deviation_grid_supply >= lower_limit_deviation:
+            charged_deviation_power = max_pos_deviation_grid_supply - lower_limit_deviation
         else:
-            max_power_grid_supply_outside_cst_flex = max(power_outside_core_standing_time_flex_list)
+            charged_deviation_power = 0.0
```

Comment:
`charged_deviation_power = max(max_pos_deviation_grid_supply - lower_limit_deviation, 0)`

Reply:
Do I understand it right, that you want to replace the if/else (line 480 - 483) by that one line you suggested? That would not be the same thing...

Reply:
Yes, that is the intention. In which case would there be a difference?
If max_deviation is lower than lower_limit, the difference is negative, so with max() the result becomes 0 (your else-branch). If it is not lower, the difference is positive or zero, max() does nothing and the result is simply the difference (your if-branch).

Reply:
I thought it through once again and I think you're right. Sorry. I will make your suggested changes.